### DISCUSSION TITLE
The great pinout sudoku [lilygo-t-display-s3]

### DISCUSSION THREAD

**Discussion by zavorra** [alf45tar/PedalinoMini#519] [Q&A]
Hi @alf45tar, hi all

Sorry for the long post. Long story short, I am asking your help to
1) build something similar to a Mini 8
2) extend it with some more button.

I am at my second build, here, but I am still not sure on how to proceed.

I have a Lilygo T-DIsplay-S3 (_well, they were two, but it turned out that the screen is really fragile..._). I started building the small pedalboard in the picture, having 2 ramps of 4 push buttons and one analog jack. 

![image](https://github.com/alf45tar/PedalinoMini/assets/12064434/4c363dcd-2b14-4f7b-a574-08b190435a0e)

Anyway, I am not satisfied because we can have on the screen the names of the pedals (in my case just 2) instead of the buttons, or the actions (see issue #434 ). 
So when @alf45tar provided the drawings for the Pedalino Mini 8 with 8 footswitches and 2 ADC I decided to go that way.  

I am trying to figure out the connections on the Mini 8 as in https://raw.githubusercontent.com/alf45tar/PedalinoMini-8/main/images/PedalinoMini%208%20Wiring.jpg, assuming that it is using a lilygo t-display s3 as per the part list

![image](https://github.com/alf45tar/PedalinoMini/assets/12064434/9075fa26-c59c-4f29-861a-5008992480ba)
 
but if I cross the board pinout with the wiring diagram and the pin definition in `Pedalino.h`, i get this

```
#elif defined ARDUINO_LILYGO_T_DISPLAY_S3   // https://github.com/Xinyuan-LilyGO/T-Display-S3
...
const byte pinD[] = {GPIO_NUM_10, GPIO_NUM_2,  GPIO_NUM_43, GPIO_NUM_44, GPIO_NUM_21, GPIO_NUM_1,  GPIO_NUM_14, GPIO_NUM_0};
const byte pinA[] = {GPIO_NUM_11, GPIO_NUM_12, GPIO_NUM_13, GPIO_NUM_16, GPIO_NUM_17, GPIO_NUM_18, GPIO_NUM_14, GPIO_NUM_0};

-------- Pedalino Mini 8 buttons vs lilygo t-display-s3 -------------

top-left

PIN NUMBER:    (PIN_18)    (PIN_17)   (PIN_43)   (PIN_21)          BUTTON ROW 1
PIN ASSIGNMENT: pinA[5]     pinA[4]     pinD[2]    pinD[4]

PIN NUMBER:    (PIN_02)   (PIN_16)   (PIN_03)   (PIN_13)           BUTTON ROW 2
PIN ASSIGNMENT: pinD[1]     pinA[3]  ?????       pinA[2]

                                                  bottom-right

```

So, my first question is: how am i supposed to wire a configuration similar to Pedalino Mini 8? Also, in the Mini 8 wiring diagram no resistors are shown, is this correct?

THEN, I would like to finish the bigger gray pedalboard with:
- 8 push buttons to send CC/PC. I wold like the names of those on the display
- 2 push buttons to navigate trough the banks
- 1 jack for the expression pedal (possibly)
- No legacy midi in/out, only BLE and USB
- No leds
- No battery

What could be the smartest wiring/configuration? I was thinking
- "Pedals 1 to 6" ->  6 momentary_1 buttons (i.e. 4 digital pin + 2 analog pin)
- "Pedal 7" -> 1 digital switch + analog ![image](https://github.com/alf45tar/PedalinoMini/assets/12064434/b94bbe3b-7227-440e-84d7-ea72d71b1c6e) (1 analog + 1 digital pin)
- "Pedal 8" -> 1 ladder with 3 switches (1 analog + 1 digital pin)
 
Is it feasible? How am I supposed to wire it?

Thanks in advance for having read all this :)

Z

**Comment by alf45tar** [comment]
The pinout overlapped is not correct because it is from the bottom of the board. If you check the opposite row it is in sync with the `Pedalino.h`.

**Comment by zavorra** [comment]
D'OH!

so if i consider this layout
![image](https://github.com/alf45tar/PedalinoMini/assets/12064434/75c4fd81-7be9-4ac1-8b43-00e91b19579b)

i get 


```
up left

** PEDAL 1 (moment_2) **    ** PEDAL 2 (moment_2) **
PIN_10      PIN_11         PIN_2       PIN_12
pinD[0]     pinA[0]        pinD[1]     pinA[1]

** PEDAL 3 (moment_2) **    ** PEDAL 4 (moment_2) **
PIN_43      PIN_13         PIN_44      PIN_16
pinD[2]     pinA[2]        pinD[3]      pinA[3]

                                    down right

**** PEDAL 6  (ADC) ****       
ADC_1 PIN_17    PIN_21 
      pinA[4]   pinD[4]

**** PEDAL 7 (ADC) ****
ADC_2  PIN_18  PIN_1
      pinA[5]   pinD[5]
   ```
which makes far more sense


Are the "PEDAL X" numbering in this table correct?

So, if I understand, you have: 4 x "Momentary 2" and 2 x "expression pedal"  PEDALS, which is cool, it has no need for resistors but still is not a workaround for issue #434 :(


Z

**Comment by alf45tar** [comment]
Pedals numbering is not correct. Pedal 6 and Pedal 7 are Pedal 5 and Pedal 6.
